### PR TITLE
Launch new project page

### PR DESCRIPTION
**Pull request by DevDelvo**
[x] - combine the look and feel of Launch Page 1, with the content of Launch Page 2
[x] - for the uploading of images - react-dropzone
     - [x] create a general reusable component for uploading a series of photos  
[x]- When you type save, make API request (multi-part form)

### CODE REVIEW

File: client/src/components/UploadImages.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    const { setImages, images } = props;
+
+    const onDrop = useCallback(acceptedFiles => {
+        setImages(acceptedFiles)
+    }, [setImages])
+
+    const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop })
+    return (
+        <React.Fragment>
+            <h3>Upload images</h3>
```
Comment (evankp) [inline client/src/components/UploadImages.js]: you can use material-ui's `<Typography variant="h3">Upload Images</Typography>` here

File: client/src/components/UploadImages.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            <div {...getRootProps()}>
+                <input {...getInputProps()} />
+                {
+                    isDragActive ?
+                        <p>Drop the files here ...</p> :
+                        <p>Drag 'n' drop some files here, or click to select files</p>
```
Comment (evankp) [inline client/src/components/UploadImages.js]: you can use material ui's Typography here with variant body1 for each `p`

File: client/src/components/UploadImages.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                    images ?
+                        images.map(image => (
+                            <img
+                                key={image.name}
+                                src={URL.createObjectURL(image)}
+                                alt=""
+                                style={{ maxHeight: '100px', maxWidth: '100px' }}
+                            />
+                        )) :
+                        <p>Select images to see preview here.</p>
```
Comment (evankp) [inline client/src/components/UploadImages.js]: Same as above comment

File: client/src/pages/AddProject.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+    }
+
+    handleSubmit = async (event) => {
+        event.preventDefault();
+        const validation = this.validators.validate(this.state);
+        this.setState({ validation });
+
+        if (validation.isValid) {
+            const { formData } = this.state;
+            const { id } = this.props.match.params; // will have to change eventually.
```
Comment (evankp) [inline client/src/pages/AddProject.js]: Not anything wrong with this line, but bookmarking it for change in later ticket/PR? Maybe: #42 if merged before

File: client/src/pages/AddProject.js [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        const { classes } = this.props;
+        const { title, subtitle, industry, location, images, fundingGoal, validation } = this.state;
+        return (
+            <main className={classes.pageContent}>
+                {/* <div className={classes.projectPreview}>
+                    <h2>Product Preview Thing</h2>
+                    <Button type="submit" variant="contained" color="primary" >
+                        Preview
+                    </Button>
+                </div> */}
```
Comment (evankp) [inline client/src/pages/AddProject.js]: Commented out code

File: client/src/pages/AddProject.js [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+                                            key={industry.id}
+                                            value={industry.name}
+                                        >
+                                            {industry.name}
+                                        </option>
+                                    )
+                                })
+                            }
+                        </OutlinedSelect>
+                        <h3>Project Location</h3>
```
Comment (evankp) [inline client/src/pages/AddProject.js]: Can use Material-ui's Typography variant h3 here

File: client/src/pages/AddProject.js [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+                                            value={location.name}
+                                        >
+                                            {location.name}
+                                        </option>
+                                    )
+                                })
+                            }
+                        </OutlinedSelect>
+                        <UploadImages setImages={this.setImages} images={images} />
+                        <h3>Funding goal amount</h3>
```
Comment (evankp) [inline client/src/pages/AddProject.js]: Can use Material-ui's Typography variant h3 here

File: client/src/pages/AddProject.js [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+        return (
+            <main className={classes.pageContent}>
+                {/* <div className={classes.projectPreview}>
+                    <h2>Product Preview Thing</h2>
+                    <Button type="submit" variant="contained" color="primary" >
+                        Preview
+                    </Button>
+                </div> */}
+                <div className={classes.addProjectPage}>
+                    <h2>Start with basics</h2>
```
Comment (evankp) [inline client/src/pages/AddProject.js]: Also I created a component for the header, as I think I recall it looking similar to login/register pages: CenteredPageHeader

**Review by evankp — CHANGES_REQUESTED** [review]
I would use material-ui's Typography component for tags like `<p>` and `<h3>`, keeps styles a little more in sync, ~~also I think you need to merge dev into this branch~~ and lastly is there going to be styling or separating that out?

File: client/src/components/UploadImages.js (hunk 1 above)
Comment (DevDelvo) [reply client/src/components/UploadImages.js]: Thanks!

File: client/src/pages/AddProject.js (hunk 8 above)
Comment (DevDelvo) [reply client/src/pages/AddProject.js]: Ah I opted not to go with it because I just wanted the text without the green border

**Review by evankp — APPROVED** [review]
A bit uneasy about merging with only one approval of two, but I'd say it's a good start, it just doesn't match quite up to specs/stuck to left

File: client/src/App.js [hunk 9]
```diff
@@ -75,7 +76,8 @@ const App = () => {
             path="/login"
             render={(routerProps) => <Login setAuthenticated={setAuthenticated} {...routerProps} />}
           />
-          <Route path="/profiles/:id" component={ProfilePage}/>
+          <Route path="/profiles/:id" component={ProfilePage} />
+          <Route path="/projects/add/:id" component={AddProject} />
```
Comment (tsyaeger) [inline client/src/App.js]: I'm curious as to why the projects/add/id format for the route. 
I would have expected something like users/id/projects, but don't know if I'm maybe missing something

File: client/src/api/projects.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import httpClient from './httpClient';
+
+export const addProject = async (id, projectData) => {
+    const url = `/api/projects/add/${id}`;
+    const token = localStorage.getItem('jwtToken')
+    const authOptions = {
```
Comment (tsyaeger) [inline client/src/api/projects.js]: The interceptor in the api should theoretically be modifying the header to contain the token if it exists in local storage.

File: client/src/api/projects.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        headers: {
+            Authorization: `Bearer ${token}`,
+        }
+    }
+    try {
+        const response = await httpClient.post(url, projectData, authOptions);
+        if (response.data.err) return { ...response.data }
+
+        if (typeof window !== 'undefined') {
```
Comment (tsyaeger) [inline client/src/api/projects.js]: No problem with this -- it's just new to me. Is there a scenario in which the window would be undefined? I think it's just the DOM here, no?

File: client/src/components/UploadImages.js [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    //     padding: '4px',
+    //     borderRadius: '10px',
+    //     cursor: 'pointer'
+    // }
+}
+
+const UploadImages = (props) => {
+    const { classes, setImages, images, deleteImage } = props;
+
+    const onDrop = useCallback(acceptedFiles => {
```
Comment (tsyaeger) [inline client/src/components/UploadImages.js]: I'm totally nitpicking but useCallback could use a more descriptive name.

File: client/src/pages/AddProject.js [hunk 13]
```diff
@@ -0,0 +1,279 @@
...
+    },
+    formLine: {
+        marginBottom: '20px',
+    },
+    button: {
+        margin: '20px 0px 10px 0px'
+    }
+}
+
+const industries = [
```
Comment (tsyaeger) [inline client/src/pages/AddProject.js]: Maybe the least pressing issue of all, but maybe these can live somewhere as constants or something like that?

File: client/src/App.js (hunk 9 above)
Comment (DevDelvo) [reply client/src/App.js]: The route is subject to change. It's probably better to do something like users/:id/projects/add or something like that

File: client/src/components/UploadImages.js (hunk 12 above)
Comment (DevDelvo) [reply client/src/components/UploadImages.js:47]: it's actually a react hook so I don't think I can do anything about it. I also used it from the example provided in the react drop-zone docs. I think useCallback() returns a memoized callback so it's more efficient when dealing with these images

File: client/src/api/projects.js (hunk 11 above)
Comment (DevDelvo) [reply client/src/api/projects.js]: I think it's just some habit I picked up while learning this stuff. I think it's for if your script is running in a web browser or not. Could remove it

File: client/src/pages/AddProject.js (hunk 13 above)
Comment (DevDelvo) [reply client/src/pages/AddProject.js]: Yeah. It's really just dummy data since I don't really know how we will handle lists of locations and industries in the future. Industries seem much more manageable but locations can get pretty huge. I'll take your suggestion and move them to a separate folder and file

File: client/src/api/projects.js (hunk 10 above)
Comment (DevDelvo) [reply client/src/api/projects.js:6]: The issue is that at times the token does not load consistently. It may be an issue with my machine but there would be times that it gave the token, but other times that it didn't. A jwtToken is consistently added to localStorage, but in the httpClient folder, there are times where it logged as null for me despite it showing up in localStorage.

**Comment by DevDelvo** [comment]
> The page looks great, but I wasn't able to create any projects and wasn't given any indication as to what, as a user, I had done wrong. So some sort of error message would be good. I couldn't create a project either with or without an image. Not sure if I didn't do something correctly.

That's odd because you should be able to create a project with or without the images since they're not required. I would make sure that you're logged in and that in the route `/projects/add/:id` that the id matches your logged in id. It may be an issue with your s3 permissions as well. It would be hard for me to make an error message for the images since they're not required so those are the two things that come to mind as to why it may not be working.

File: client/src/api/projects.js (hunk 10 above)
Comment (tsyaeger) [reply client/src/api/projects.js]: strange. Thanks for letting me know. I'll create an issue for it.

File: client/src/api/projects.js (hunk 11 above)
Comment (tsyaeger) [reply client/src/api/projects.js]: Interesting. Yeah, I'm not sure. Is there a time when it wouldn't be running in a browser?

File: client/src/components/UploadImages.js (hunk 12 above)
Comment (tsyaeger) [reply client/src/components/UploadImages.js]: Got it. I'm just going to go ahead and relay my feedback to the drop-zone people...

File: client/src/pages/AddProject.js (hunk 13 above)
Comment (tsyaeger) [reply client/src/pages/AddProject.js]: okay -- I thought they were constants. Good point about handling location. That will definitely be an interesting challenge.

**Comment by tsyaeger** [comment]
I had to change the add project controller to allow project creation without images in postman. Even after I did that the page wouldn't work (I'll try to adjust permissions etc) but I nonetheless think the user should see some basic error message.

File: client/src/api/projects.js (hunk 11 above)
Comment (DevDelvo) [reply client/src/api/projects.js]: I think the only times would be if it's running in node or with web workers. I could remove it for this project; I don't think it would be a big deal

File: client/src/components/UploadImages.js (hunk 12 above)
Comment (DevDelvo) [reply client/src/components/UploadImages.js:47]: it's from the react hook devs! loll

File: client/src/pages/AddProject.js (hunk 13 above)
Comment (DevDelvo) [reply client/src/pages/AddProject.js]: Yeah I'm curious to see how it's handled. I imagine it's just some object with locations or maybe some crazy data structure I haven't heard of yet lol

File: client/src/App.js (hunk 9 above)
Comment (tsyaeger) [reply client/src/App.js]: Maybe. It would definitely be more intuitive. For instance, at first glance I had no idea what the :id actually referred to in the path. Could maybe even just do "/projects" because the project is created by the current user?

**Comment by DevDelvo** [comment]
I'm having an issue where I am not logged in when I refresh the page. I have my JWT_SECRET and REACT_APP_JWT_SECRET matching and the token is definitely place into localStorage upon logging in. Otherwise, the launch new project page is working fine

**Comment by evankp** [comment]
@DevDelvo I'll check this out on my end tonight and see if I can replicate it.

File: client/src/api/projects.js (hunk 10 above)
Comment (evankp) [reply client/src/api/projects.js:6]: @DevDelvo I know localstorage.getItem() returns null if the key is not found in localstorage, but 🤔

File: client/src/api/projects.js (hunk 11 above)
Comment (evankp) [reply client/src/api/projects.js]: As @DevDelvo probably not a big deal, as this won't be running in node or non-browsers

File: client/src/api/projects.js [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+            let projects = [];
+
+            if (localStorage.getItem('projects')) {
+                projects = JSON.parse(localStorage.getItem('projects'));
+            }
+            projects.push(response.data);
+
+            localStorage.setItem('projects', JSON.stringify(projects))
```
Comment (evankp) [inline client/src/api/projects.js]: maybe we should consider making a global state in app.js instead of local? idk, I wouldn't think of storing projects in local, but I guess for caching?

File: client/src/components/Image.js [hunk 15]
```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import { withStyles } from '@material-ui/core'
+import DeleteOutlinedIcon from '@material-ui/icons/DeleteOutlined';
+
+const styles = {
+  deleteIconContainer: {
+    position: 'absolute',
+    height: '75px',
+    width: '75px',
```
Comment (evankp) [inline client/src/components/Image.js:9]: You can just simply put 75 as a js number here, no need for px.

File: client/src/components/Image.js [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+    backgroundColor: 'grey',
+    opacity: '50%',
+    borderRadius: '10px',
```
Comment (evankp) [inline client/src/components/Image.js:12]: Same as above, any px can simply be put as a js number, I think it processes it as a px unit by default, so that will probably apply to any values that are similar in the following code.

File: client/src/components/Image.js [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+    setHovering(toggleHoverState)
+  }
+
+  const toggleHoverState = () => {
+    setHovering(!isHovering)
+  }
+
+  // const opacityStyle = () => {
+  //   return isHovering ? { opacity: '75%'} : { opacity: '100%' };
+  // }
```
Comment (evankp) [inline client/src/components/Image.js]: You can set hover styles in the styles/css by doing something like the following, and :hover is a css selector so in css you could do `h1:hover {...}`, and in css-in-js something like: `'&:hover': {opacity: '75%'}`

File: client/src/components/Image.js [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+      {
+        isHovering && <div className={classes.deleteIconContainer}><DeleteOutlinedIcon className={classes.deleteIcon} /></div>
+      }
+      <img
+        onClick={() => deleteImage(image)}
+        className={classes.image}
+        key={image.name}
+        src={URL.createObjectURL(image)}
+        alt=""
+        // style={opacityStyle()}
```
Comment (evankp) [inline client/src/components/Image.js]: Commented out code? 🤷‍♂

File: client/src/components/Inputs.js [hunk 19]
```diff
@@ -14,10 +14,9 @@ const OutlinedSelect = (props) => {
 
     return (
         <React.Fragment>
-            <InputLabel ref={inputLabel} htmlFor={props.selectId} {...props.inputProps}>{props.labelText}</InputLabel>
+            <InputLabel ref={inputLabel} htmlFor={props.selectId} {...props.inputProps}><Typography variant="h4">{props.labelText}</Typography></InputLabel>
             <Select
                 native
-                value={props.value}
```
Comment (evankp) [inline client/src/components/Inputs.js:20]: Why did you remove this, in React arn't inputs controlled by state...?

File: client/src/dummyData/dropDownItems.js [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+    { id: 123, name: 'New York' },
+    { id: 431, name: 'California' },
+    { id: 903, name: 'Georgia' },
+    { id: 223, name: 'Florida' }
+]
+
+export {
+    industries,
+    locations
+}
```
Comment (evankp) [inline client/src/dummyData/dropDownItems.js:21]: Honestly dunno about putting dummydata in separate folder, as I tend to put it directly in state before hooking it up, but it doesn't matter that much to block anything

File: client/src/pages/AddProject.js [hunk 21]
```diff
@@ -0,0 +1,251 @@
...
+const styles = {
+    pageContent: {
+        display: 'flex',
+        justifyContent: 'left'
+    },
+    projectPreviewContainer: {
+        display: 'flex',
+        width: '25%',
+        flexDirection: 'column',
+        margin: '0px 20px 0px 10px',
```
Comment (evankp) [inline client/src/pages/AddProject.js]: As a suggestion, You could do something simple like `marginRight: 20` and `marginLeft: 10` instead here

**Comment by evankp** [comment]
@DevDelvo try running `npm ci` (like npm install but reinstalls packages without modifying package-lock or new versions. Maybe there is an issue with you missing a package? runs fine on my system

File: client/src/components/Inputs.js (hunk 19 above)
Comment (DevDelvo) [reply client/src/components/Inputs.js:20]: I have no idea why this was removed either. It's back now

File: client/src/api/projects.js (hunk 14 above)
Comment (DevDelvo) [reply client/src/api/projects.js]: Yeah that's probably to just have the projects in a global state that initializes with a getProjects API call?

File: client/src/components/Image.js (hunk 17 above)
Comment (DevDelvo) [reply client/src/components/Image.js]: I'm going to keep the logic of my hovering since I use it to display a DeletedOutlinedIcon from material-ui

Comment (evankp) [reply client/src/components/Image.js]: DeletedOutlinedIcon?

Comment (DevDelvo) [reply client/src/components/Image.js]: Yeah!

File: client/src/pages/AddProject.js [hunk 22]
```diff
@@ -0,0 +1,253 @@
...
+const styles = {
+    pageContent: {
+        display: 'flex',
+        justifyContent: 'left'
+    },
+    projectPreviewContainer: {
+        display: 'flex',
+        width: '25%',
+        flexDirection: 'column',
+        // margin: '0px 20px 0px 10px',
```
Comment (evankp) [inline client/src/pages/AddProject.js]: Commented out code

File: client/src/pages/AddProject.js (hunk 4 above)
Comment (evankp) [reply client/src/pages/AddProject.js]: fyi, could get this from props.userDetails object now that the PR has been merged

**Review by evankp — APPROVED** [review]
Most of my other comments that are remaining still stand, but also just some minor details that we can change later if needed, so not blocking.
